docs/api-reference/service-status-api.md

```diff
@@ -23,154 +23,1015 @@ sidebar_label: Service status
 
-Similar to `/druid/historical/v1/loadstatus`, but instead of returning JSON with a flag, responses 200 OK if segments
-in the local cache have been loaded, and 503 SERVICE UNAVAILABLE, if they haven't.
+<!--END_DOCUSAURUS_CODE_TABS-->
 
+#### Sample request
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--cURL-->
+```shell
+curl "http://<HISTORICAL_IP>:<HISTORICAL_PORT>/druid/historical/v1/loadstatus"
+```
+<!--HTTP-->
+```http
+GET /druid/historical/v1/loadstatus HTTP/1.1
+Host: http://<HISTORICAL_IP>:<HISTORICAL_PORT>
+```
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+#### Sample response
+<details>
+  <summary>Click to show sample response</summary>
+  
+  ```json
+  {
+    "cacheInitialized": true
+  }
+  ```
+</details>
+
+### Get segment readiness
+
+#### URL
+<code class="getAPI">GET</code> `/druid/historical/v1/readiness`
+
+Retrieves a status code to indicate if all segments in the local cache have been loaded. Similar to `/druid/historical/v1/loadstatus`, but instead of returning JSON with a flag, it returns status codes.
+
+#### Responses
+<!--DOCUSAURUS_CODE_TABS-->
+
+<!--200 SUCCESS-->
+<br/>
+*Segments in local cache successfully loaded*  
+<!--503 SERVICE UNAVAILABLE-->
+<br/>
+*Segments in local cache have not been loaded*  
+
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+#### Sample request
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--cURL-->
+```shell
+curl "http://<HISTORICAL_IP>:<HISTORICAL_PORT>/druid/historical/v1/readiness"
+```
+<!--HTTP-->
+```http
+GET /druid/historical/v1/readiness HTTP/1.1
+Host: http://<HISTORICAL_IP>:<HISTORICAL_PORT>
+```
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+#### Sample response
+A successful response to this endpoint results in an empty response body.
 
 ## Load Status
 
-`GET /druid/broker/v1/loadstatus`
+### Get Broker query load status
+
+#### URL
+
+<code class="getAPI">GET</code> `/druid/broker/v1/loadstatus`
+
+Retrieves a flag indicating if the Broker knows about all segments in the cluster. 
+
+Use this endpoint to know when a Broker service is ready to be queried after a restart.
+
+#### Responses
+<!--DOCUSAURUS_CODE_TABS-->
+
+<!--200 SUCCESS-->
+<br/>
+*Segments successfully loaded*  
+
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+#### Sample request
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--cURL-->
+```shell
+curl "http://<BROKER_IP>:<BROKER_PORT>/druid/broker/v1/loadstatus"
+```
+<!--HTTP-->
+```http
+GET /druid/broker/v1/loadstatus HTTP/1.1
+Host: http://<BROKER_IP>:<BROKER_PORT>
+```
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+#### Sample response
+<details>
+  <summary>Click to show sample response</summary>
+  ```json
+  {
+    "inventoryInitialized": true
+  }
+  ```
+</details>
+
+### Get Broker query readiness
+
+#### URL
+
+<code class="getAPI">GET</code> `/druid/broker/v1/readiness`
+
+Retrieves a status code indicating if the Broker knows about all segments in the cluster and is ready to be queried after a restart. Similar to `/druid/broker/v1/loadstatus`, but instead of returning a JSON, it returns status codes.
```

Comment:
```suggestion
Retrieves a status code indicating if the Broker knows about all segments in the cluster and is ready to accept queries after a restart. Similar to `/druid/broker/v1/loadstatus`, but instead of returning a JSON, it returns status codes.
```
nit passive. "status code indicating if the Broker knows about all segments" reads awkwardly, but I can't think of a better way to reword it.